server/service.go

```diff
@@ -28,24 +33,58 @@ func newService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) *service
 }
 
 type services struct {
-	svcMap     map[string]*service
-	defaultSvc *service
+	svcSearchMap                       map[string]*service // key: "svcName.methodName" and "methodName", value: svcInfo
+	svcMap                             map[string]*service // key: service name, value: svcInfo
+	conflictingMethodHasFallbackSvcMap map[string]bool
+	fallbackSvc                        *service
+	hasCombineService                  bool
 }
 
 func newServices() *services {
-	return &services{svcMap: map[string]*service{}}
+	return &services{
+		svcSearchMap:                       map[string]*service{},
+		svcMap:                             map[string]*service{},
+		conflictingMethodHasFallbackSvcMap: map[string]bool{},
+	}
 }
 
-func (s *services) addService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) {
+func (s *services) addService(svcInfo *serviceinfo.ServiceInfo, handler interface{}, isFallbackService bool) error {
 	svc := newService(svcInfo, handler)
-	if s.defaultSvc == nil {
-		s.defaultSvc = svc
+
+	// combine service check
+	if svcInfo.ServiceName == "CombineService" {
+		s.hasCombineService = true
+	} else if s.hasCombineService {
```

Comment:
how about the combine service is the last register service? 
1. you should also check if the svcMap is empty when combineService is registered?
2. I think you may don't need the hasCombineService, just check if there is combineService in svcMap